middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -908,9 +908,21 @@ let effects_and_coeffects_of_unary_primitive p =
   | Int_as_pointer -> Effects.No_effects, Coeffects.No_coeffects
   | Opaque_identity -> Effects.Arbitrary_effects, Coeffects.Has_coeffects
   | Int_arith (_, (Neg | Swap_byte_endianness))
-  | Num_conv _ | Boolean_not | Reinterpret_int64_as_float
-  | Float_arith (Abs | Neg) ->
+  | Num_conv _ | Boolean_not | Reinterpret_int64_as_float ->
     Effects.No_effects, Coeffects.No_coeffects
+  | Float_arith (Abs | Neg) ->
+    (* Float operations are not really pure since they actually access the
+       globally mutable rounding mode, which can be changed (but only from C
+       code). The Flambda_features.float_const_prop tracks whether we are
+       allowed to make optimizations assuming a non-changing rounding mode (i.e.
+       'float_const_prop () = true' means that the rounding should not be
+       changed by user code, and thus float optimizations are allowed).
+       Therefore, when 'float_const_prop () = false', we add coeffects to float
+       operations so that they cannot be moved through an effectful operation.
+       (e.g. a call to a c stub that changes the rounding mode). *)
```

Comment:
You could add a cross-reference to the `eligible_for_cse` functions too.

Reply:
done